.github/workflows/generate-dependencies.js

```diff
@@ -0,0 +1,134 @@
+const fs = require('fs');
+const path = require('path');
+
+const ignoredFolders = ['.github', '.gradle', '.idea', '.git', 'node_modules', 'lib', 'src', 'tests'];
+const generatedFileName = 'provenance.json';
+const projectName = 'vcd-ext-sdk/';
+
+async function generateFileHelper(currentDir, components) {
+    try {
+        const files = await fs.promises.readdir(currentDir);
+
+        for (const file of files) {
+            const filePath = path.join(currentDir, file);
+            if (file === 'package-lock.json') {
+                const packageLockJson = await readFile(filePath);
+                const packageJsonPath = filePath.replace('package-lock.json', 'package.json');
+                const packageJson = await readFile(packageJsonPath);
+                addDependencies(packageLockJson, packageJson.version, components, currentDir);
+            } else {
+                // Stat the file to see if we have a file or dir
+                const stat = await fs.promises.stat(filePath);
+
+                if (stat.isDirectory() && !ignoredFolders.includes(file)) {
+                    await generateFileHelper(filePath, components);
+                }
+            }
+        }
+    } catch (e) {
+        console.error("Could not generate dependencies for project.", e);
+    }
+}
+
+async function readFile(filePath) {
```

Comment:
If I call this function without taking a look at it, I would expect that it returns a raw file content.